core/controllers/topics_and_skills_dashboard.py

```diff
@@ -174,6 +177,16 @@ def post(self):
             state_domain.SubtitledHtml.from_dict(explanation_dict))
         skill_services.save_new_skill(self.user_id, skill)
 
+        for filename in filenames:
```

Comment:
Should we call skill.get_all_image_filenames() and `self.payload.get('filenames')`

Reply:
I don't quite understand, why would I call skill.get_all_image_filenames()?

Reply:
skill.get_all_image_filenames will return all the names of the images used in the skill and then we can validate whether we are saving all and only required images in the GCS.

Reply:
Done